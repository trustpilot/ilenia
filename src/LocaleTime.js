import PropTypes from 'prop-types';
import { useTranslations } from './';

const LocaleTime = ({ date, format }) => {
  const [, locale] = useTranslations();
  const dateWrapper = new Date(date);

  if (isNaN(dateWrapper)) {
    // eslint-disable-next-line no-console
    console.error('Invalid date');
    return null;
  }

  try {
    return dateWrapper.toLocaleTimeString(locale, format);
  } catch (err) {
    return dateWrapper.toLocaleTimeString('en-US', format);
  }
};

LocaleTime.propTypes = {
  date: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.instanceOf(Date)]),
  format: PropTypes.shape({
    weekday: PropTypes.oneOf(['narrow', 'short', 'long']),
    era: PropTypes.oneOf(['narrow', 'short', 'long']),
    year: PropTypes.oneOf(['numeric', '2-digit']),
    month: PropTypes.oneOf(['numeric', '2-digit', 'narrow', 'short', 'long']),
    day: PropTypes.oneOf(['numeric', '2-digit']),
    hour: PropTypes.oneOf(['numeric', '2-digit']),
    minute: PropTypes.oneOf(['numeric', '2-digit']),
    second: PropTypes.oneOf(['numeric', '2-digit']),
    timeZoneName: PropTypes.oneOf(['short', 'long']),
  }),
};

export default LocaleTime;
