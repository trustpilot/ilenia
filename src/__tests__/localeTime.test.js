import React from 'react';
import renderer from 'react-test-renderer';

import LocalizationProvider from '../LocalizationProvider';
import LocaleTime from '../LocaleTime';

test('Renders a localized time from a string', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleTime date={'2018-07-10T12:38:37.713'} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a localized time from a date', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleTime date={new Date('2018-07-10T12:38:37.713')} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders null if the date is invalid', () => {
  // eslint-disable-next-line no-console
  console.error = jest.fn();

  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleTime date={'abcdefghijklmnopqrstuvwxyz'} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // eslint-disable-next-line no-console
  expect(console.error).toHaveBeenCalled();
});

test('Renders a localized time with an incorrect locale', () => {
  const component = renderer.create(
    <LocalizationProvider locale="abcdefghijklmnopqrstuvwxyz" translations={{}}>
      <LocaleTime date={'2018-07-10T12:38:37.713'} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a localized time with specific format', () => {
  const component = renderer.create(
    <LocalizationProvider locale="abcdefghijklmnopqrstuvwxyz" translations={{}}>
      <LocaleTime
        date="2018-07-10T12:38:37.713"
        format={{
          hour: 'numeric',
          minute: 'numeric',
        }}
      />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
