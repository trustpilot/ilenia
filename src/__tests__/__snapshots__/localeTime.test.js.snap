// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a localized time from a date 1`] = `12:38:37 PM`;

exports[`Renders a localized time from a string 1`] = `12:38:37 PM`;

exports[`Renders a localized time with an incorrect locale 1`] = `12:38:37 PM`;

exports[`Renders a localized time with specific format 1`] = `12:38 PM`;

exports[`Renders null if the date is invalid 1`] = `null`;
